pegasus/helpers/form_helpers.rb

```diff
@@ -135,7 +135,15 @@ def insert_form(kind, data, options={})
     updated_ip: request.ip,
   }
   row[:user_id] = dashboard_user[:id] if dashboard_user
-  row[:id] = DB[:forms].insert(row)
+
+  if kind == "HocSignup2015"
+    update_row = DB[:forms].where(email: row[:email], kind: kind, name: row[:name]).update(row)
```

Comment:
one more thing with this line -- it changes the return value of this method such that the row returned does not have the id. Is that ok with the callers?